File: 0/Templates/VisualBasic/ProjectTemplates/ContentAppVBTemplate/Presentation/BooleanToVisibilityConverter.vb
Namespace Presentation

    Public Class BooleanToVisibilityConverter
        Implements IValueConverter

        Private _inverse As Boolean
        Public Property Inverse As Boolean
            Get
                Return _inverse
            End Get
            Set(value As Boolean)
                _inverse = value
            End Set
        End Property

        Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
            Dim bValue = DirectCast(value, Boolean)
            If Inverse Then
                bValue = Not bValue
            End If
            If bValue Then
                Return Visibility.Visible
            End If
            Return Visibility.Collapsed
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
            Dim result As Boolean = value = Visibility.Visible
            If Inverse Then
                result = Not result
            End If

            Return result
        End Function
    End Class
End Namespace

File: 0/Templates/CSharp/ProjectTemplates/ContentAppTemplate/Presentation/BooleanToVisibilityConverter.cs
using System;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;

namespace $safeprojectname$.Presentation
{
    public class BooleanToVisibilityConverter
        : IValueConverter
    {
        public bool Inverse { get; set; }

        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var bValue = (bool)value;
            if (this.Inverse) {
                bValue = !bValue;
            }
            return bValue ? Visibility.Visible : Visibility.Collapsed;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            var result = (Visibility)value == Visibility.Visible;
            if (this.Inverse) {
                result = !result;
            }
            return result;
        }
    }
}
